docs/book/src/reference/attributes.md

```diff
@@ -0,0 +1,52 @@
+# Attributes
+
+The Sway compiler supports a list of attributes that perform various operations that are useful for building, testing and documenting libraries. Below is a list of all available attributes:
+
+## Allow
+
+The `#[allow(dead_code)]` attribute overrides the check for dead code so that violations will go unreported.
+
+## Doc
+
+The `#[doc(..)]` attribute specifies documentation.
+
+Line doc comments beginning with exactly three slashes `///`, are interpreted as a special syntax for doc attributes. That is, they are equivalent to writing `#[doc("...")]` around the body of the comment, i.e., `/// Foo` turns into `#[doc("Foo")]`
+
+Line comments beginning with `//!` are doc comments that apply to the parent of the comment, rather than the item that follows. That is, they are equivalent to writing `#![doc("...")]` around the body of the comment. `//!` comments are usually used to document modules that occupy a source file.
```

Comment:
cc @eureka-cpu - is this an accurate description of how things work in Sway?

Reply:
So it's not that they are usually used, but they can _only_ be used for documenting modules.

Reply:
Removed the part about `//!` and `#![doc("...")]`.

Reply:
Should we completely remove them completely or simply limit their scope to modules?

Reply:
I think we should include it as @mohammadfawaz suggested

Reply:
If you do, it would also close #4114

Reply:
Added again the `//!` part, with scope limited to modules and mentioned that it should be used at the beginning of Sway files.